presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkRowOutputOperator.java

```diff
@@ -107,6 +123,31 @@ public int getPartition(Page page, int position)
         }
     }
 
+    private static class PreDeterminedPartitionFunction
+            implements PartitionFunction
+    {
+        private final int partitionId;
+
+        public PreDeterminedPartitionFunction(int partitionId)
+        {
+            this.partitionId = partitionId;
+        }
+
+        @Override
+        public int getPartitionCount()
+        {
+            // To be compatible with RowIndex in PrestoSparkRowBatchBuilder
+            // where RowIndex requires an array size of at least partitionId + 1
+            return partitionId + 1;
```

Comment:
This looks a little bit like a hack. Should we set it to real number of the output partitions (`hashPartitionCount`)?

Reply:
Yes, we can use `hashPartitionCount`, the performance overhead should be negligible.